components/TimelineFilter.js

```diff
@@ -0,0 +1,56 @@
+import styled from "styled-components";
+
+export default function TimelineFilter({
+  selectedTimeframe,
+  onTimeframeChange,
+}) {
+  const timeframes = [
+    { label: "7 Days", value: 7 },
+    { label: "30 Days", value: 30 },
+    { label: "Half Year", value: 180 },
+    { label: "Full Year", value: 365 },
+  ];
```

Comment:
not really an issue, but this data never changes but is recreated with every render. if you move it out of the function body it is only generated once and reused for every render.